queryweaver_sdk/connection.py

```
@@ -0,0 +1,138 @@
+                return FalkorDB(connection_pool=self._pool)
+            except Exception as e:
+                raise ConnectionError(f"Failed to connect to FalkorDB with URL: {e}") from e
+
+        # Fall back to host/port
+        host = self._host or os.getenv("FALKORDB_HOST", "localhost")
+        port = self._port or int(os.getenv("FALKORDB_PORT", "6379"))
+
+        try:
+            return FalkorDB(host=host, port=port)
+        except Exception as e:
+            raise ConnectionError(f"Failed to connect to FalkorDB at {host}:{port}: {e}") from e
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)